docs/src/modules/java/pages/eventsourced.adoc

```diff
@@ -134,48 +134,49 @@ By default, the generated constructor has an `EventSourcedEntityContext` paramet
+The following snippet shows how the `ShoppingCartTestKit` is used to test the `ShoppingCart` implementation. AkkaServerless provides two main APIs for unit tests, the `ShoppingCartTestKit` and the `EventSourcedResult`. The former gives us the overall state of the entity. Its state and all the events over the course of all the RPC calls. While the latter only holds the effects produced for each individual RPC call to the Entity.
 
-The following is a simplification of its contents but in essence, what we show in the next snippet is the signature of the available methods
-
-.target/generated-test-sources/com/example/shoppingcart/domain/ShoppingCartTestKit.java
+.src/test/java/com/example/shoppingcart/domain/ShoppingCartTest.java
 [source,java]
 ----
-include::example$testkit/java-eventsourced-shopping-cart/intarget/generated-test-sources/akkaserverless/com/example/shoppingcart/domain/ShoppingCartTestKit.java[] 
+include::example$java-eventsourced-shopping-cart/src/test/java/com/example/shoppingcart/domain/ShoppingCartTest.java[] 
 ----
-<1> This constructor requires a `ShoppingCart` Entity so we can interact with this Entity through the `ShoppingCartTestKit`.
-<2> This constructor allows us to set a state on the Entity, so we can start the test directly from the desired state instead of having to send all the commands that would produce such a state.
-<3> This method returns the current state of the Entity. 
-<4> This method returns the accumulated events from all the executed commands sent to the entity.
-<5> This method calls to the method with the same name in the Event Source Entity passed in the constructor.
+<1> creating the TestKit passing the constructor of the Entity.
+<2> calling the method `addItem` from the Entity in the `ShoppingCartTestKit`.
+<3> calling the method `addItem` from the Entity in the `ShoppingCartTestKit`.
+<4> checking the `EventSourcedResult` of the first call to `addItem`.
+<5> checking the `EventSourcedResult` of all the calls to `addItem`.
+<6> retrieving the first event generated from the first call to `addItem`.
+<7> retrieving the second event generated from the first call to `addItem`. There is no such event as our implementation only generates one event when addItem it's called.
+<8> retrieving the response from the call to `addItem`.
+<9> retrieving the state of the entity after all the calls to `addItem`.
+
 
+*EventSourcedResult*
 
-Once we call a  method on the entity this gives us back a link:{attachmentsdir}/api/com/akkaserverless/javasdk/testkit/EventSourcedResult.html[`EventSourcedResult` {tab-icon}, window="new"]. This class has methods that we can use to assert our expectations, such as:
+Calling an entity method through the TestKit gives us back an link:{attachmentsdir}/api/com/akkaserverless/javasdk/testkit/EventSourcedResult.html[`EventSourcedResult` {tab-icon}, window="new"]. This class has methods that we can use to assert our expectations, such as:
```

Comment:
shouldn't it be "a link"?

Reply:
I'm not sure what "link" is referring to here. "gives us back a result:" would read better?

Reply:
ops! was a typo, sorry. `link` was meant to be part of the attachment so it would read "gives us back an EventSourceResult`". Fixed now.